src/canvas-player-controls.js

```diff
@@ -0,0 +1,115 @@
+import videojs from 'video.js';
+/**
+ * This class reacts to interactions with the canvas and
+ * triggers appropriate functionality on the player. Right now
+ * it does two things:
+ *
+ * 1. A `mousedown`/`touchstart` followed by `touchend`/`mouseup` without any
+ *    `touchmove` or `mousemove` toggles play/pause on the player
+ * 2. Only moving on/clicking the control bar or toggling play/pause should
+ *    show the control bar. Moving around the scene in the canvas should not.
+ */
+class CanvasPlayerControls extends videojs.EventTarget {
+  constructor(player, canvas) {
+    super();
+
+    this.player = player;
+    this.canvas = canvas;
+
+    this.onMoveEnd = videojs.bind(this, this.onMoveEnd);
+    this.onMoveStart = videojs.bind(this, this.onMoveStart);
+    this.onMove = videojs.bind(this, this.onMove);
+    this.onControlBarMove = videojs.bind(this, this.onControlBarMove);
+
+    this.player.controlBar.on([
+      'mousedown',
+      'mousemove',
+      'mouseup',
+      'touchstart',
+      'touchmove',
+      'touchend'
+    ], this.onControlBarMove);
+
+    // we have to override these here because
+    // video.js listens for user activity on the video element
+    // and makes the user active when the mouse moves.
+    // We don't want that for 3d videos
+    this.oldReportUserActivity = this.player.reportUserActivity;
```

Comment:
not a huge fan of these but I don't think there's currently a way to disable it. Maybe we should have one?
Though, I guess this is also fairly self-contained and restores it on dispose.

Reply:
Yeah I thought about adding one, but it seems like this plugin would be the only use case.